cpp/include/cudf/strings/string_view.inl

```diff
@@ -84,6 +88,16 @@ __device__ inline size_type string_view::length() const
 {
     if( _length <= UNKNOWN_STRING_LENGTH )
         _length = strings::detail::characters_in_string(_data,_bytes);
+    if( _length && (_char_width <= UNKNOWN_CHAR_WIDTH) )
```

Comment:
Why `<=` ?

```suggestion
    if( _length && (_char_width == UNKNOWN_CHAR_WIDTH) )
```

Reply:
Bit flipping :-)
https://www.wnycstudios.org/podcasts/radiolab/articles/bit-flip
https://www.livescience.com/8170-toyota-recall-caused-cosmic-rays.html
https://www.schneier.com/blog/archives/2015/03/hardware_bit-fl.html
Also, I was following this pattern:
https://github.com/rapidsai/cudf/blob/162aa73eaf898d19b636a7338b7b2c4ec4b09db4/cpp/src/column/column.cu#L138

Reply:
:)
Haha, but my comments on slack on bit flipping were in the context of the termination condition in a for loop with an increasing counter. Comparing to a specific value is different. I think a bit flip here is bad no matter what. I would worry about this causing a bug if another sentinel/flag value were defined in the future. (Hadn't noticed the `UNKNOWN_NULL_COUNT` code.)